### PR TITLE
Issue #14631: Updated HTML_ELEMENT in JavadocTokenTypes.java  to new AST format

### PR DESCRIPTION
**Pull request by souvikrajsingh**
Issue: #14631

**Command used:**

` java -jar checkstyle-10.21.0-all.jar -J Test.java | sed "s/\[[0-9]\+:[0-9]\+\]//g"`

**Test.java**
```
/**
 * @throws IOException if <b>connection</b> problems occur
 */
public class Test {
}
```

```
Souvik@souviks MINGW64 ~/Desktop/checkStylee/HTML Element AST
$ java -jar checkstyle-10.21.0-all.jar -J Test.java | sed "s/\[[0-9]\+:[0-9]\+\]//g"
COMPILATION_UNIT -> COMPILATION_UNIT
`--CLASS_DEF -> CLASS_DEF
    |--MODIFIERS -> MODIFIERS
    |   |--BLOCK_COMMENT_BEGIN -> /*
    |   |   |--COMMENT_CONTENT -> *\r\n * @throws IOException if <b>connection</b> problems occur\r\n
    |   |   |   `--JAVADOC -> JAVADOC
    |   |   |       |--NEWLINE -> \r\n
    |   |   |       |--LEADING_ASTERISK ->  *
    |   |   |       |--WS ->
    |   |   |       |--JAVADOC_TAG -> JAVADOC_TAG
    |   |   |       |   |--THROWS_LITERAL -> @throws
    |   |   |       |   |--WS ->
    |   |   |       |   |--CLASS_NAME -> IOException
    |   |   |       |   |--WS ->
    |   |   |       |   `--DESCRIPTION -> DESCRIPTION
    |   |   |       |       |--TEXT -> if
    |   |   |       |       |--HTML_ELEMENT -> HTML_ELEMENT
    |   |   |       |       |   `--HTML_TAG -> HTML_TAG
    |   |   |       |       |       |--HTML_ELEMENT_START -> HTML_ELEMENT_START
    |   |   |       |       |       |   |--START -> <
    |   |   |       |       |       |   |--HTML_TAG_NAME -> b
    |   |   |       |       |       |   `--END -> >
    |   |   |       |       |       |--TEXT -> connection
    |   |   |       |       |       `--HTML_ELEMENT_END -> HTML_ELEMENT_END
    |   |   |       |       |           |--START -> <
    |   |   |       |       |           |--SLASH -> /
    |   |   |       |       |           |--HTML_TAG_NAME -> b
    |   |   |       |       |           `--END -> >
    |   |   |       |       |--TEXT ->  problems occur
    |   |   |       |       |--NEWLINE -> \r\n
    |   |   |       |       `--TEXT ->
    |   |   |       `--EOF -> <EOF>
    |   |   `--BLOCK_COMMENT_END -> */
    |   `--LITERAL_PUBLIC -> public
    |--LITERAL_CLASS -> class
    |--IDENT -> Test
    `--OBJBLOCK -> OBJBLOCK
        |--LCURLY -> {
        `--RCURLY -> }
```

### CODE REVIEW

**Comment by romani** [comment]
please read https://ci.appveyor.com/project/Checkstyle/checkstyle/builds/51235642/job/h37llqj15md2jp5f#L111 and fix it.

**Comment by souvikrajsingh** [comment]
@romani This is now fixed, thanks.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 1]
```diff
@@ -1397,26 +1397,26 @@ public final class JavadocTokenTypes {
...
-     *                        |--END[1x25] : [>]
-     *                    |--TEXT[1x26] : [connection]
-     *                    |--HTML_ELEMENT_END[1x36] : [</b>]
-     *                        |--START[1x36] : [<]
-     *                        |--SLASH[1x37] : [/]
-     *                        |--HTML_TAG_NAME[1x38] : [b]
-     *                        |--END[1x39] : [>]
-     *            |--TEXT[1x40] : [ problems occur]
+     * {@code |--JAVADOC_TAG -> JAVADOC_TAG
+     *        |--THROWS_LITERAL -> @throws
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:1401]: Escaping is required for `>` please look at others javadocs that already has new format

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:1402]: still not done.

Please read and watch videos at https://checkstyle.org/beginning_development.html#Starting_Development
them show exactly on how to do this update

Comment (souvikrajsingh) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:1402]: Sure thanks @romani  for letting me know, will check and fix it !

**Comment by romani** [comment]
please watch videos, it shows how to do single commit.

**Comment by souvikrajsingh** [comment]
@romani Hi thank you for taking the time to review my PR and give feedback
 I have made the necessary changes as you asked!
 
  Kindly let me know if I missed something

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 2]
```diff
@@ -1397,26 +1397,27 @@ public final class JavadocTokenTypes {
...
+     *                        |--START -&gt; <
+     *                        |--HTML_TAG_NAME -&gt; b
+     *                        `--END -&gt; >
+     *                    |--TEXT -> connection
+     *                    `--HTML_ELEMENT_END -&gt; HTML_ELEMENT_END
+     *                        |--START -&gt; <
+     *                        |--SLASH -&gt; /
+     *                        |--HTML_TAG_NAME -&gt; b
+     *                        `--END -&gt; >
+     *            |--TEXT ->  problems occur
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:1420]: Please restore `|` to this node

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 3]
```diff
@@ -1397,26 +1397,27 @@ public final class JavadocTokenTypes {
...
+     *        |--CLASS_NAME -&gt; IOException
+     *        |--WS -&gt;
+     *        `--DESCRIPTION -&gt; DESCRIPTION
+     *            |--TEXT -&gt; if
+     *            |--HTML_ELEMENT -&gt; HTML_ELEMENT
+     *                `--HTML_TAG -&gt; HTML_TAG
+     *                    |--HTML_ELEMENT_START -&gt; HTML_ELEMENT_START
+     *                        |--START -&gt; <
+     *                        |--HTML_TAG_NAME -&gt; b
+     *                        `--END -&gt; >
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:1413]: `<` and `>` should be escaped

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java (hunk 3 above)
Comment (souvikrajsingh) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:1413]: Thanks again how can i miss that ! 

@romani on it

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java (hunk 2 above)
Comment (souvikrajsingh) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:1419]: i am not sure if i understood what you meant here, can you please tell me which node? 


> Please restore `|` to this node

**Comment by romani** [comment]
Single commit, always , please.

**Comment by souvikrajsingh** [comment]
> Single commit, always , please.

It is now in single commit, i would request you to review again! thank you for your patience, its my first contribution!

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java (hunk 2 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: This node is hanging without branch.

This is AST, tree, all should be connected by pseudo graphics symbols.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]:
```
|
|
|
```

**Comment by romani** [comment]
```
    |   |   |       |--JAVADOC_TAG -> JAVADOC_TAG
    |   |   |       |   |--THROWS_LITERAL -> @throws
    |   |   |       |   |--WS ->
    |   |   |       |   |--CLASS_NAME -> IOException
    |   |   |       |   |--WS ->
    |   |   |       |   `--DESCRIPTION -> DESCRIPTION
    |   |   |       |       |--TEXT -> if
    |   |   |       |       |--HTML_ELEMENT -> HTML_ELEMENT
    |   |   |       |       |   `--HTML_TAG -> HTML_TAG
    |   |   |       |       |       |--HTML_ELEMENT_START -> HTML_ELEMENT_START
    |   |   |       |       |       |   |--START -> <
    |   |   |       |       |       |   |--HTML_TAG_NAME -> b
    |   |   |       |       |       |   `--END -> >
    |   |   |       |       |       |--TEXT -> connection
    |   |   |       |       |       `--HTML_ELEMENT_END -> HTML_ELEMENT_END
    |   |   |       |       |           |--START -> <
    |   |   |       |       |           |--SLASH -> /
    |   |   |       |       |           |--HTML_TAG_NAME -> b
    |   |   |       |       |           `--END -> >
    |   |   |       |       |--TEXT ->  problems occur
    |   |   |       |       |--NEWLINE -> \r\n
    |   |   |       |       `--TEXT ->
```

We need line from HTML_ELEMENT

**Comment by souvikrajsingh** [comment]
> ```
>     |   |   |       |--JAVADOC_TAG -> JAVADOC_TAG
>     |   |   |       |   |--THROWS_LITERAL -> @throws
>     |   |   |       |   |--WS ->
>     |   |   |       |   |--CLASS_NAME -> IOException
>     |   |   |       |   |--WS ->
>     |   |   |       |   `--DESCRIPTION -> DESCRIPTION
>     |   |   |       |       |--TEXT -> if
>     |   |   |       |       |--HTML_ELEMENT -> HTML_ELEMENT
>     |   |   |       |       |   `--HTML_TAG -> HTML_TAG
>     |   |   |       |       |       |--HTML_ELEMENT_START -> HTML_ELEMENT_START
>     |   |   |       |       |       |   |--START -> <
>     |   |   |       |       |       |   |--HTML_TAG_NAME -> b
>     |   |   |       |       |       |   `--END -> >
>     |   |   |       |       |       |--TEXT -> connection
>     |   |   |       |       |       `--HTML_ELEMENT_END -> HTML_ELEMENT_END
>     |   |   |       |       |           |--START -> <
>     |   |   |       |       |           |--SLASH -> /
>     |   |   |       |       |           |--HTML_TAG_NAME -> b
>     |   |   |       |       |           `--END -> >
>     |   |   |       |       |--TEXT ->  problems occur
>     |   |   |       |       |--NEWLINE -> \r\n
>     |   |   |       |       `--TEXT ->
> ```
> 
> We need line from HTML_ELEMENT

than you for the reviewing  this again, will get this done asap!

**Comment by souvikrajsingh** [comment]
@romani I have update the PR, kindly review and let me know if further changes are needed. Thanks!

**Review by romani — APPROVED** [review]
Thanks a lot.
Try to fix one more AST , to be sure we are on same page.

**Comment by souvikrajsingh** [comment]
> Thanks a lot. Try to fix one more AST , to be sure we are on same page.

Thank you for your patience with my PR, it was my first , will surely start working on this ASAP!

**Pull request merged**